src/test/runTest.ts

```diff
@@ -17,12 +17,29 @@ async function main() {
         const extensionTestsPath: string = path.join(__dirname, 'index');
 
         // Download VS Code, unzip it and run the integration tests
-        await runTests({
-            version: 'insiders',
-            extensionDevelopmentPath,
-            extensionTestsPath,
-            launchArgs: ['--disable-extensions', '-n', targetResourcesDir]
-        } as TestOptions);
+        const maxRetries: number = 3;
+        let retryCount: number = 0;
+        
+        while (retryCount < maxRetries) {
+          try {
+            let exitCode: number = await runTests({
+              version: 'insiders',
+              extensionDevelopmentPath,
+              extensionTestsPath,
+              launchArgs: ['--disable-extensions', '-n', targetResourcesDir]
+            } as TestOptions);
+            console.log(`Test execution (runTests) finished with exit code: ${exitCode}.`);
+             // No error occurred, exit the loop
+            break;
+          } catch (error: any) {
+            if (error.toString().includes('read ECONN') || (error.message && error.message.includes('read ECONN'))) {
```

Comment:
I'm not sure why, but `npm run format` removes the "any" from the error.
Let's do this change and run `npm run format`.
```suggestion
                if ((<Error>error).toString().includes('read ECONN') || (<Error>error)?.message.includes('read ECONN')) {
```